diff --git a/flexkvs/server/ialloc.c b/flexkvs/server/ialloc.c
index 9d08801..4cea47f 100644
--- a/flexkvs/server/ialloc.c
+++ b/flexkvs/server/ialloc.c
@@ -86,7 +86,8 @@ void ialloc_init(void)
         mem_base_phys = id.base;
     }
 #else
-    if ((seg_base = mmap(NULL, total, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB | MAP_POPULATE,
+    // if ((seg_base = mmap(NULL, total, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB | MAP_POPULATE,
+    if ((seg_base = mmap(NULL, total, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_POPULATE,
                     -1, 0)) == MAP_FAILED)
     //if ((seg_base = mmap(NULL, total, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB,
     //                -1, 0)) == MAP_FAILED)
