diff --git a/flexkvs/bench/commandline.c b/flexkvs/bench/commandline.c
index 07f3728..4bc7678 100644
--- a/flexkvs/bench/commandline.c
+++ b/flexkvs/bench/commandline.c
@@ -276,7 +276,7 @@ int parse_settings(int argc, char *argv[], struct settings *s)
                 total_size = strtoull(optarg, &end, 0);
                 // Ratio: 1:31 hashtable:segments
                 s->hasht_size = total_size / 32;
-                s->segmaxnum = total_size * 31 / (32 * s->segsize);
+                s->segmaxnum = 7 + total_size * 31 / (32 * s->segsize);
                 hash_size = ((double)s->hasht_size / (1024.0 * 1024.0 * 1024.0));
                 alloc_size = ((double)s->segmaxnum * (double)s->segsize / (1024.0 * 1024.0 * 1024.0));
 
diff --git a/flexkvs/server/ialloc.c b/flexkvs/server/ialloc.c
index 9d08801..4cea47f 100644
--- a/flexkvs/server/ialloc.c
+++ b/flexkvs/server/ialloc.c
@@ -86,7 +86,8 @@ void ialloc_init(void)
         mem_base_phys = id.base;
     }
 #else
-    if ((seg_base = mmap(NULL, total, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB | MAP_POPULATE,
+    // if ((seg_base = mmap(NULL, total, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB | MAP_POPULATE,
+    if ((seg_base = mmap(NULL, total, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_POPULATE,
                     -1, 0)) == MAP_FAILED)
     //if ((seg_base = mmap(NULL, total, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB,
     //                -1, 0)) == MAP_FAILED)
